### PR TITLE
nsncd: unstable-2024-03-18 -> unstable-2024-08-06

### PR DESCRIPTION
**Pull request by picnoir**
## Description of changes

:warning:  Note: this is kind of blocking the next staging merge! We rollbacked by accident Nsncd in staging to a previous version with known bugs.

Continuation of https://github.com/NixOS/nixpkgs/pull/332176#issuecomment-2270824319

Currently picking the commit from a upstream open PR https://github.com/twosigma/nsncd/pull/119. If upstream end up merging the PR fast enough, it'd make sense to wait for it to be merged.

I'll open a subsequent PR to Nixpkgs to point to the merged PR if they're too slow for us.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by alyssais** [comment]
If upstream is slow, I would rather revert that commit and hold off with the Rust upgrade than depend on fetching a commit that's liable to disappear in future when GitHub GCs.

**Comment by alyssais** [comment]
Let's start by reverting, anyway, because we definitely don't want to end up with the downgrade: https://github.com/NixOS/nixpkgs/pull/332692.

**Comment by picnoir** [comment]
@alyssais PTAL

File: pkgs/os-specific/linux/nsncd/default.nix [hunk 1]
```diff
@@ -2,21 +2,27 @@
...
     hash = "sha256-Bd7qE9MP5coBCkr70TdoJfwYhQpdrn/zmN4KoARcaMI=";
   };
 
-  cargoHash = "sha256-i1rmc5wxtc631hZy2oM4d6r7od0w8GrG7+/pdM6Gqco=";
+  cargoHash = "sha256-06fbzx+rqO4k1Cm/UaLk5SRs1wWPSpOG/qCie4c/nPQ=";
+
+  # TOREMOVE when https://github.com/twosigma/nsncd/pull/119 gets merged.
+  cargoPatches = [
+    ./0001-cargo-bump.patch
+    ./0002-ignored-unused-hostent-field.patch
```
Comment (alyssais) [inline pkgs/os-specific/linux/nsncd/default.nix:23]: Might be nicer to avoid a vendored patch with

```nix
RUSTFLAGS = "-A dead_code";
```

Comment (picnoir) [reply pkgs/os-specific/linux/nsncd/default.nix]: done

**Review by alyssais — APPROVED** [review]

**Review by flokli — APPROVED** [review]

**Comment by alyssais** [comment]
Oh, commit message is not in the right format.  I'll just fix that with Squash and Merge.

**Comment by picnoir** [comment]
Duh, sorry about that. Forgot to check what was the previous version…

File: pkgs/os-specific/linux/nsncd/default.nix [hunk 2]
```diff
@@ -2,21 +2,29 @@
...
 , fetchFromGitHub
 , rustPlatform
-, nix-gitignore
 }:
 
-rustPlatform.buildRustPackage rec {
+rustPlatform.buildRustPackage {
   pname = "nsncd";
-  version = "unstable-2024-03-18";
+  version = "unstable-2024-08-06";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/nsncd/default.nix:9]:
```suggestion
  version = "1.4.1-unstable-2024-04-10";
```
that is supposed to be the date of the commit

Comment (picnoir) [reply pkgs/os-specific/linux/nsncd/default.nix:9]: The commit introduced through the patch has been authored today.

See https://github.com/twosigma/nsncd/pull/119

Comment (alyssais) [reply pkgs/os-specific/linux/nsncd/default.nix:9]: We don't usually consider non-upstream patches to affect the version.

Comment (alyssais) [reply pkgs/os-specific/linux/nsncd/default.nix:9]: We don't usually consider non-upstream patches to affect the version.

**Review by vcunat — CHANGES_REQUESTED** [review]
As noted by OfBorg, this PR won't build:
```
error: hash mismatch in fixed-output derivation '/nix/store/2ykd2ikw45jp1p5ci56q8z5zmrr567pn-nsncd-1.4.1-unstable-2024-04-10-vendor.tar.gz.drv':
         specified: sha256-06fbzx+rqO4k1Cm/UaLk5SRs1wWPSpOG/qCie4c/nPQ=
            got:    sha256-N7U9YsyGh8+fLT973GGZTmVXcdnWhpqkeYTxzJ0rzdo=
```

**Comment by picnoir** [comment]
Changing the program version changes the cargo vendor hash. TIL.

**Pull request merged**